### PR TITLE
devtools: enables typescript `strict` option

### PR DESCRIPTION
**Pull request by JeanMeche**
Enabling `strict` is part of an effort to improve the quality of the devtools code base. 
One of the direct side effect is to enable `noImplicitAny`, `strictPropertyInitialization` and `strictBindCallApply`.

This commit also replaces `fullTemplateTypeCheck` with `stringTemplates`.

cc @ducin

### CODE REVIEW

File: devtools/projects/ng-devtools-backend/src/lib/directive-forest/ltree.ts [hunk 1]
```diff
@@ -37,7 +41,7 @@ const isLView = (value: any): boolean => {
 };
 
 export const METADATA_PROPERTY_NAME = '__ngContext__';
-export const getLViewFromDirectiveOrElementInstance = (dir: any): null|{} => {
+export function getLViewFromDirectiveOrElementInstance(dir: any): null|LView {
```
Comment (kbrilla) [inline devtools/projects/ng-devtools-backend/src/lib/directive-forest/ltree.ts:44]: any reason this is converted to function while on line 35 `isLContainer`  is kept as const?

File: devtools/projects/ng-devtools-backend/src/lib/state-serializer/state-serializer.ts [hunk 2]
```diff
@@ -51,38 +51,39 @@ const nestedSerializerContinuation = (nodes: NestedProp[], level: number) =>
...
-      return result;
-    };
+export function serializeDirectiveState(
+    instance: object, levels = MAX_LEVEL): {[key: string]: Descriptor} {
+  const result: Record<string, Descriptor> = {};
+  getKeys(instance).forEach((prop) => {
+    if (typeof prop === 'string' && ignoreList.has(prop)) {
+      return;
+    }
+    // TODO: investigate why this call didn't match the signature
```
Comment (JeanMeche) [inline devtools/projects/ng-devtools-backend/src/lib/state-serializer/state-serializer.ts:78]: This one is particularly suspicious.

**Review by AleksanderBodurri — APPROVED** [review]
Awesome stuff! Thanks @JeanMeche

**Comment by dylhunn** [comment]
We discussed, and Alex's approval should be sufficient

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 6cd91c675c8d248736cfe447f2a5d6dc488093b8.

**Pull request closed without merging**